modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -187,24 +194,95 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
+   */
+  private void cleanMySqlCleanupThreadsThreadFactory(Class<?> cleanupThreadsClass)
+      throws NoSuchFieldException, IllegalAccessException {
+    // In new mysql driver versions (at least 8), the executor service is wrapped inside a delegate class
+    // (DelegatedExecutorService) that exposes only the ExecutorService interface. In order to clean the threadPoolExecutor
+    // classloader reference, it has to be extracted manually though reflection from each delegate/wrapper class.
+    // Hierarchy leading to real ThreadPoolExecutor is: AbandonedConnectionCleanupThread.cleanupThreadExecutorService ->
+    // class DelegatedExecutorService.e -> class ThreadPoolExecutor.
+    // Note that the field 'cleanupThreadExcecutorService' is mispelled. There's actually a typo in MySql driver code.
+    Field cleanupExecutorServiceField = cleanupThreadsClass
+        .getDeclaredField("cleanupThreadExcecutorService");
+    cleanupExecutorServiceField.setAccessible(true);
+    ExecutorService delegateCleanupExecutorService =
+        (ExecutorService) cleanupExecutorServiceField.get(cleanupThreadsClass);
+
+    Field realExecutorServiceField = delegateCleanupExecutorService.getClass().getSuperclass().getDeclaredField("e");
+    realExecutorServiceField.setAccessible(true);
+    ThreadPoolExecutor realExecutorService =
+        (ThreadPoolExecutor) realExecutorServiceField.get(delegateCleanupExecutorService);
+
+    realExecutorService.setThreadFactory(Executors.defaultThreadFactory());
+  }
+
+  /**
+   * Sends a shutdown message to MySql's connection cleanup thread class.
+   * 
+   * @param classAbandonedConnectionCleanupThread
+   * @throws IllegalAccessException
+   * @throws InvocationTargetException
+   * @throws NoSuchMethodException
+   */
+  private void shutdownMySqlConnectionCleanupThreads(Class<?> classAbandonedConnectionCleanupThread)
+      throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+    try {
+      Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
+      uncheckedShutdown.invoke(null);
+    } catch (NoSuchMethodException e) {
+      Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
+      checkedShutdown.invoke(null);
+    }
+  }
+
+  /**
+   * Tries to find the MySql driver AbandonedConnectionCleanupThread class, with the known class addresses.
+   * 
+   * @return The MySql driver AbandonedConnectionCleanupThread class object, if found.
+   */
+  private Class<?> findMySqlDriverClass() {
+    Class<?> foundClass = null;
+    for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+      try {
+        foundClass =
+            this.getClass().getClassLoader().loadClass(knownCleanupThreadClassAddress);
+      } catch (ClassNotFoundException e) {
+        logger.warn("No AbandonedConnectionCleanupThread registered with class address " + knownCleanupThreadClassAddress);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Format specifiers should be used instead of string concatenation. ([squid:S3457](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3457))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAu&open=AWjzLS16otIXnmsUYkAu)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAu

Reply:
Debug level log, otherwise we'll be logging a warning even if the second class was the one used by the driver.
Also, as sonar says, use the logger formatting: 
```
logger.debug("No AbandonedConnectionCleanupThread registered with class address {}", knownCleanupThreadClassAddress);
```